src/network/clientpackethandler.cpp

```diff
@@ -228,29 +227,23 @@ void Client::handleCommand_AccessDenied(NetworkPacket* pkt)
 	u8 denyCode;
 	*pkt >> denyCode;
 
-	if (denyCode == SERVER_ACCESSDENIED_SHUTDOWN ||
-			denyCode == SERVER_ACCESSDENIED_CRASH) {
+	if (pkt->getRemainingBytes() > 0)
 		*pkt >> m_access_denied_reason;
-		if (m_access_denied_reason.empty())
+
+	if (m_access_denied_reason.empty()) {
+		if (denyCode != SERVER_ACCESSDENIED_CUSTOM_STRING) {
 			m_access_denied_reason = gettext(accessDeniedStrings[denyCode]);
+		} else if (denyCode >= SERVER_ACCESSDENIED_MAX) {
```

Comment:
this needs to come first or the bounds check is broken

Reply:
Done in 304a89e34